### ISSUE TITLE
Add Weight column to inventory screen

### ISSUE THREAD

**Issue by koolraap** [NetHack/NetHack#310]
I'd like to add an option to have a weight column on the inventory screen. I implemented this ages ago in a long-lost branch, I'm happy to do so again if the PR will get accepted.

My rationale is it's nuts not knowing what weighs what, your character must know, why not have some hints? Especially when encumbered, it'd be nice knowing which suit of armor weighs the most.

I'll make it optional, off by default. My lost code didn't account for hallucination, I guess it should.

Any thoughts?

**Comment by copperwater** [comment]
I've currently got pull request #280 open for this (it doesn't try to format it as a column, but otherwise pretty much the same). Doesn't account for hallucination either, but I don't think it should, because hallucination doesn't obscure the appearances of items in your inventory.

**Comment by koolraap** [comment]
Excellent! I'll have to give your PR a try. It looks a bit different to what I remember (I definitely didn't consider wizard mode) but it's been a long time (a decade?) since I was poking around.

Re Hallucination: I was thinking it'd be fun to get a list of wacky units of mass and use those. It'd be accurate but useless :-)

**Comment by kristjanlink** [comment]
Perhaps it would be a good idea to implement weights in a way that isn't direct (like add it to enlightenment or another spell/potion/wand/etc.) because like in real life you can't accurately tell the weight of an item. Perhaps have the values rounded somehow or have the weights be approximate by being off by some (randomized?) margin depending on an attribute, wisdom would make sense (like shop prices depend on charisma).

**Comment by copperwater** [comment]
Non-uniform weights don't play nice with item stacking, because an item stack has a single weight field. (What happens if a weight 8 rock tries to merge with a weight 11 rock? And then what happens when you drop one of the rocks?)

And if actual item weights are uniform, but the UI fuzzes and misrepresents them to the player, then that's just an interface screw. You'll still be able to look in objects.c or the wiki to find the real weights, and it will be better just to turn off the lying weight display.

The reason for openly displaying accurate, uniform weights is that they're only really hidden for unspoiled players, and the gameplay benefit of being able to see weights (like being more informed to decide on a choice of armor like @koolraap said) outweighs keeping them hidden for the questionable benefit of unspoiled players.

**Comment by nhmall** [comment]
Moved to [Feature Requests](https://github.com/NetHack/NetHack/wiki/Feature-Requests) page, with a link back to this Issue and its comments.

Closing the issue.
